lib/widgets/forms/task_form.dart

```diff
@@ -0,0 +1,556 @@
+				Future.wait([
+					Future(widget.createTaskCallback(task))
+				]);
+			else
+				Future.wait([
+					Future(widget.saveTaskCallback(task))
+				]);
+			Navigator.of(context).pop();
+		}
+	}
+
+	Future<bool> exitForm(BuildContext context, bool isSystemPop) {
+		if (!isDataChanged) {
+			Navigator.pop(context, true);
+			return Future.value(false);
+		} else {
+			FocusManager.instance.primaryFocus.unfocus();
+			return showDialog<bool>(
+				context: context,
+				builder: (c) => AlertDialog(
+					title: Text(AppLocales.of(context).translate('alert.unsavedProgressTitle')),
+					content: Text(AppLocales.of(context).translate('alert.unsavedProgressMessage')),
+					actions: [
+						FlatButton(
+							child: Text(AppLocales.of(context).translate('actions.cancel')),
+							onPressed: () => Navigator.pop(c, false),
+						),
+						FlatButton(
+							textColor: Colors.red,
+							child: Text(AppLocales.of(context).translate('actions.exit')),
+							onPressed: () {
+								if(isSystemPop)
+									Navigator.pop(c, true);
+								else {
+									Navigator.of(context).pop();
+									Navigator.of(context).pop();
+								}
+							}
+						)
+					]
+				)
+			);
+		}
+	}
+
+	Widget buildCustomHeader(BuildContext context) {
+		bool hasTitle = !formModeIsCreate() && task.title != null && task.title.length > 0;
+		double appBarVerticalPadding = hasTitle ? 8.0 : 12.0;
+		return Material(
+			elevation: 4.0,
+			color: AppColors.formColor,
+			child: Container(
+				child: SafeArea(
+					child: ListTile(
+						dense: true,
+						contentPadding: EdgeInsets.symmetric(vertical: 0.0, horizontal: 4.0),
+						trailing: HelpIconButton(helpPage: 'task_creation'),
+						leading: IconButton(
+							tooltip: AppLocales.of(context).translate('actions.back'),
+							icon: Icon(Icons.arrow_back, color: Colors.white),
+							onPressed: () => exitForm(context, false),
+						),
+						title: Padding(
+							padding: EdgeInsets.only(top: appBarVerticalPadding, bottom: appBarVerticalPadding, left: 4.0),
+							child: Column(
+								crossAxisAlignment: CrossAxisAlignment.start,
+								children: <Widget>[
+									AnimatedDefaultTextStyle(
+										duration: Duration(milliseconds: 200),
+										child: Text(AppLocales.of(context).translate('$_pageKeyPlanForm.${formModeIsCreate() ? 'addTaskButton' : 'editTaskButton'}')), 
+										style: hasTitle ?
+											Theme.of(context).textTheme.bodyText1 :
+											Theme.of(context).textTheme.headline3.copyWith(color: Colors.white, fontSize: 20.0)
+									),
+									AnimatedSwitcher(
+										duration: Duration(milliseconds: 400),
+										transitionBuilder: (child, animation) {
+											return SizeTransition(
+												sizeFactor: animation,
+												axisAlignment: 1,
+												axis: Axis.vertical,
+												child: child
+											);
+										},
+										child: hasTitle ?
+											Hero(
+												tag: formModeIsCreate() ? "none3463634634" : widget.task.key,
```

Comment:
Is that an angry-typed random string? :smile:

Reply:
Yes! :D That way I know it's not suppose to connect to anything. 


_or maybe it's some secret code? 🤔_